nix/commands.nix

```diff
@@ -1,44 +1,98 @@
 { pkgs, unstable, ... }:
   let
+    bash  = "${pkgs.bash}/bin/bash";
     ssh   = "${pkgs.openssh}/bin/ssh";
     stack = "${unstable.stack}/bin/stack";
     ghcid = "${pkgs.ghcid}/bin/ghcid";
 
-    task = name: body:
+    cmd = description: script:
+      { inherit description;
+        inherit script;
+      };
+
+    command = {name, script, description ? "<No description given>"}:
       let
         package =
           pkgs.writeScriptBin name ''
-            #!${pkgs.stdenv.shell}
+            #!${bash}
             echo "⚙️  Running ${name}..."
             unset STACK_IN_NIX_SHELL
-            ${body}
+            ${script}
           '';
 
         bin = "${package}/bin/${name}";
       in
-         { package = package;
-           bin     = bin;
+         { package     = package;
+           description = description;
+           bin         = bin;
          };
 
-    build    = task "build" "${stack} build";
-    runtests = task "runtests" "${stack} test";
-    repl     = task "repl" "${stack} repl --no-nix-pure";
+    commands = defs:
+      let
+        names =
+          builtins.attrNames defs;
+
+        helper =
+          let
+            lengths = map builtins.stringLength names;
+            maxLen  = builtins.foldl' (acc: x: if x > acc then x else acc) 0 lengths;
+            maxPad  =
+              let
+                go = acc:
+                  if builtins.stringLength acc >= maxLen
+                  then acc
+                  else go (" " + acc);
+              in
+                go "";
+
+            folder = acc: name:
+              let
+                nameLen = builtins.stringLength name;
+                padLen  = maxLen - nameLen;
+                padding = builtins.substring 0 padLen maxPad;
+              in
+                acc + " && echo '${name} ${padding}| ${(builtins.getAttr name defs).description}'";
+
+            lines =
+              builtins.foldl' folder "echo ''" names;
+
+          in
+            pkgs.writeScriptBin "helpme" ''
```

Comment:
any reason for `helpme` vs. just `help` ?

Reply:
`help` is an existing command in `fish`

<img width="1419" alt="Screen Shot 2021-03-01 at 9 52 35 AM" src="https://user-images.githubusercontent.com/1052016/109537572-df3bd480-7a73-11eb-8f3b-a3cea45f02b3.png">